pkg/kubectl/cmd/util/factory.go

```diff
@@ -346,6 +351,15 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		},
 		Describer: func(mapping *meta.RESTMapping) (kubectl.Describer, error) {
 			mappingVersion := mapping.GroupVersionKind.GroupVersion()
+			if mapping.GroupVersionKind.Group == federation.GroupName {
```

Comment:
Another way to implement this is to have a way for kubectl to detect that it is talking to federation apiserver (using discovery api) and then always use the federation clientset and FederationDescriber.
That way, we can implement different federation describer for `kubectl describe services` as well.